trestle/common/list_utils.py

```diff
@@ -14,16 +14,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Trestle List Utils."""
-from typing import Callable, Dict, List, Optional
+from typing import Any, Callable, Dict, List, Optional
 
 from trestle.common.common_types import TG, TG2
+from trestle.common.err import TrestleError
 
 
 def as_list(list_or_none: Optional[List[TG]]) -> List[TG]:
     """Convert list or None object to itself or an empty list if none."""
     return list_or_none if list_or_none else []
 
 
+def comma_sep_to_list(string_or_none: Optional[str]) -> List[str]:
+    """Convert optional comma-sep string to list of stings and strip."""
+    return list(map(str.strip, string_or_none.split(','))) if string_or_none is not None else []
+
+
+def comma_colon_sep_to_dict(string_or_none: Optional[str]) -> Dict[str, str]:
+    """Convert optional comma and colong sep list to dict."""
+    entries = comma_sep_to_list(string_or_none)
+    # this is awkward to do with dict comprehension so using loop
+    dic = {}
+    for entry in entries:
+        # if more than one colon include any colons in the value after the first one
+        token = entry.split(':', 1)
+        if len(token) == 1:
+            dic[token[0].strip()] = token[0].strip()
+        else:
+            dic[token[0].strip()] = token[1].strip()
+    return dic
```

Comment:
Please see my comment in the tests about construction of dict of dictionaries...

Reply:
I believe this is a useful way to do it.  This is used as a simple form of string mapping - not for the creation of deep dictionaries.  The mapped string may contain colons but it may not contain commas.